modules/KoreBuild.Tasks/Policies/Lineup/PackageVersionSource.cs

```diff
@@ -70,33 +71,51 @@ public void AddPackagesFromFolder(string folder)
                     continue;
                 }
 
-                AddPackage(package, folder);
+                AddPackage(package, NuGetFramework.AnyFramework, folder);
                 _logger.LogVerbose($"Using package {package.Id} {package.Version} from {packageFile}");
             }
         }
 
-        public bool TryGetPackageVersion(string id, out string version)
+        public bool TryGetPackageVersion(string id, NuGetFramework targetFramework, out string version)
         {
-            version = null;
-            if (_packageLineupMap.TryGetValue(id, out var info))
+            if (!_packageLineupMap.TryGetValue(id, out var versions))
             {
-                version = info.identity.Version.ToNormalizedString();
-                return true;
+                version = null;
+                return false;
             }
 
-            return false;
+            var lineupFramework = NuGetFrameworkUtility.GetNearest(versions.Keys, targetFramework, f => f);
```

Comment:
Is `GetNearest` the right thing to use here? I'd expect we would have exact TFMs. For instance, we wouldn't want an app targeting `netcoreapp2.2` to pick up `M.N.A 2.1.0` simply because it's the closest available TFM.

Reply:
You bring up a good point. My thinking was this: there are two ways to handle TFMs from the lineup package. As an example, let's say we want PackageA 2.0.0 for .NET Framework projects, but otherwise use PackageA 1.0.0.

(1) exact TFM must match
Pros: tighter control, more predictable behavior. 
Cons: you must explicitly specify every package for every possible TFM that will ever be used. The lineup file becomes harder to author and contains lots of duplicate info.
```xml
    <dependencies>
      <group targetFramework=".NETStandard2.0">
        <dependency id="PackageA" version="1.0.0" />
      </group>
      <group targetFramework=".NETCore2.0">
        <dependency id="PackageA" version="1.0.0" />
      </group>
      <group targetFramework=".NETCore2.1">
        <dependency id="PackageA" version="1.0.0" />
      </group>
      <group targetFramework=".NETFramework4.6">
        <dependency id="PackageA" version="2.0.0" />
      </group>
      <group targetFramework=".NETFramework4.6.1">
        <dependency id="PackageA" version="2.0.0" />
      </group>
      <group targetFramework=".NETFramework4.6.2">
        <dependency id="PackageA" version="2.0.0" />
      </group>
      <group targetFramework=".NETFramework4.7">
        <dependency id="PackageA" version="2.0.0" />
      </group>
    </dependencies>
```

(2) `GetNearest`
**Pros:** we can use `AnyFramework` for most dependencies. Future TFMs follow existing compatibility fallback.
**Cons:** As you identified, there is a possibilty that when new TFMs are introduced, that fallback is wrong.
```xml
    <dependencies>
      <group>
        <dependency id="PackageA" version="1.0.0" />
      </group>
      <group targetFramework=".NETFramework4.6">
        <dependency id="PackageA" version="2.0.0" />
      </group>
    </dependencies>
```

Reply:
So far, I've chosen `GetNearest` as I think the `netcoreapp2.2` problem is easier to solve than the authoring problem of exact TFMs only.

Reply:
I guess if you're optimizing for authoring ease, sure. I'm not entirely sure why you chose nuspec as a file format which boxes you in to this sort of design.

Reply:
> I'm not entirely sure why you chose nuspec as a file format which boxes you in to this sort of design.

We considered other formats briefly in design meeting, but none of them offered obvious benefit over using NuGet's package metadata. That said, I'm not dismissing the idea. Lineups are not set in stone yet. I'd love to hear new ideas for how to implement this better.

Reply:
Maybe a mapping of TFM ranges per package? But I don't have enough context so I'll let trust your choice here 👍

Reply:
That's a good point. We don't have a way to express TFM ranges in nuspec metadata. If we had a new format, we could add those kinds of rules.

I'm trying to 🔮 a little and think if there are other behaviors we might want that we can't do with nuget metadata. I can't think of anything, but I'll add this to a new issue so we can discuss more.

Reply:
#384